File: NET/RestSharpTests/RestSharpTests/Base/BaseClass.cs
using NUnit.Framework;
using RestSharp;
using System;
using System.Collections.Generic;
using System.IO;
using System.Reflection;
using AventStack.ExtentReports;
using AventStack.ExtentReports.Gherkin.Model;
using RestSharpTests.Utilties;

namespace RestSharpTests.Base
{
    public class BaseClass : ExtentReportListener
    {


        public JsonHandler JHandler;
        public static int ClassCounter = 0;

        public RestSharpUtility restSharpUtility;

        public RestRequest restRequest;
        public RestResponse restResponse;
      
        public BaseClass()
        {

        }

        public void SetUp()
        {
            var env = TestContext.Parameters["env"];
            if (env != null)
            {
                Environment.SetEnvironmentVariable("env", env.ToLower());
            }
            else
            {
                Environment.SetEnvironmentVariable("env", string.Empty);
            }
        }

        [OneTimeSetUp]
        public void BeforeClass()
        {
            SetUp();
            SetUpReports(TestContext.CurrentContext);
            restSharpUtility = new RestSharpUtility(ObjectHelper.BASEURI);

            // Instance of other required classes
            if (TestContext.CurrentContext.Test.ClassName.ToLower().Contains("testsuite_sample"))
            {
                if (!JsonHandler.IsInitialized())
                    JHandler = new JsonHandler(new FileInfo(ObjectHelper.GetSampleTestDataFile));
                else
                    JHandler = new JsonHandler();
            }
            ClassCounter++;
        }

        [SetUp]
        public void BeforeMethod()
        {
            // Start test
            TestStarted();

        }

        [TearDown]
        public void AfterMethod()
        { 
            TestFinished();
        }

        [OneTimeTearDown]
        public void TearDown()
        {
            // Push results to report after each test
            Reports.Flush();
        }


    }
}

File: untitled/src/main/java/org/example/BaseClass.java
package org.example;

import groovyjarjarpicocli.CommandLine;
import io.restassured.specification.RequestSpecification;
import org.testng.ITestContext;
import org.testng.annotations.*;

import java.io.File;
import java.lang.reflect.Method;
import java.util.LinkedHashMap;

import static org.example.ObjectHelper.getSampleTestDataFile;

@Listeners(ExtentReportListener.class)
public class BaseClass extends ExtentReportListener{
    public LinkedHashMap<String,String> testData;
    public JsonHandler jhandler;
    public static int classCounter =0;
    public RequestSpecification ReqSpecs;

    public RestAssuredUtils RaUtils;
    @Parameters({"env"})
    @BeforeSuite
    public void setUp(@Optional String env){
        if (env !=null){
            System.setProperty("env",env.toLowerCase());
        } else {
            System.setProperty("env","");
        }
    }

    @BeforeClass
    public void beforeClass(ITestContext iTestContext){
        RaUtils = new RestAssuredUtils(org.example.ObjectHelper.BASEURI);
        //instance of other required classes
        if(iTestContext.getCurrentXmlTest().getClasses().get(classCounter).toString().contains("testSuite_Sample")) {
            jhandler = new JsonHandler(new File(org.example.ObjectHelper.getSampleTestDataFile));
       // }//else if(iTestContext.getCurrentXmlTest().getClasses().get(classCounter).toString().contains("testSuite_TF_622")){
            //jhandler = new JsonHandler(new File(ObjectHelper.getTMF622TestDataFile));

        } else {
          //  jhandler = new JsonHandler(new File(ObjectHelper.getTMF622TestDataFile));

        }
        classCounter++;
    }

    @BeforeMethod
    public void beforeMethod(Method m) {
        // Start test
        test = reports.createTest(m.getName());
    }
    @AfterSuite
    public void tearDown() {
        // push results to report after each test
        reports.flush();
    }
}

